### PR TITLE
improve project submit page

### LINKED ISSUE #228: [REFACTOR] improve project submit page
### Description

Requirement:

* improve project submit page code.
* Create input component and etc and reuse there.
* There is some error logs in console, please fix it (Seen in local development)
* Optimise code

### PR DESCRIPTION
**Pull request by rohitdasu**
### Fixes Issue
closes #228 

### Changes proposed

* console error fix
* added yup validation
* divided page into components - made reusable

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +40,7 @@
     "react-redux": "^8.0.2",
     "swr": "^1.3.0",
     "usehooks-ts": "^2.6.0",
+    "yup": "^0.32.11",
```
Comment (leewan-09) [inline package.json:43]: Why use `yup`, we have `zod`, I think we should use one package

**Review by leewan-09 — CHANGES_REQUESTED** [review]

**Review by leewan-09 — CHANGES_REQUESTED** [review]
Can we fix this problem? 

When we add 5 tags. it should show one error message. It keeps adding error messages as we click.
I think when we have an error displaced we need to reset the current error toast and show the new one 

image

File: components/SubmitPageForm/SubmitPageForm.tsx [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+      toast.success('project added successfully');
+      reset();
+      setFileError(false);
+      setFileKey(uuidv4().toString());
+    } catch (e) {
+      toast.error(e.message);
+    }
+  };
+
+  const onTagAddition = (
```
Comment (leewan-09) [inline components/SubmitPageForm/SubmitPageForm.tsx:90]: The text format should consistent.

image

**Review by leewan-09 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (rohitdasu) [reply package.json:43]: ok will need to see how we can integrate `zod` with `react-hook-form`

File: components/SubmitPageForm/SubmitPageForm.tsx (hunk 2 above)
Comment (rohitdasu) [reply components/SubmitPageForm/SubmitPageForm.tsx:90]: okay 👍

**Comment by rohitdasu** [comment]
> Can we fix this problem?
> 
> When we add 5 tags. it should show one error message. It keeps adding error messages as we click. I think when we have an error displaced we need to reset the current error toast and show the new one
> 
> image

i think this is the standard way of displaying error toast.

File: package.json (hunk 1 above)
Comment (rohitdasu) [reply package.json:43]: got it. it's similar implementation just like yup

Comment (rohitdasu) [reply package.json]: done

File: components/SubmitPageForm/SubmitPageForm.tsx (hunk 2 above)
Comment (rohitdasu) [reply components/SubmitPageForm/SubmitPageForm.tsx:90]: done

**Comment by leewan-09** [comment]
> > Can we fix this problem?
> > When we add 5 tags. it should show one error message. It keeps adding error messages as we click. I think when we have an error displaced we need to reset the current error toast and show the new one
> > image
> 
> i think this is the standard way of displaying error toast.


Is that so 🤔 . What do you say @Yazdun  @JakubChorzepa

**Review by leewan-09 — CHANGES_REQUESTED** [review]
1. Uploading an image is optional, right? I can't submit the form without uploading the image.
2. Also could you fix the text-overflow, we have lots of free space. 
image

**Comment by rohitdasu** [comment]
> 1. Uploading an image is optional, right? I can't submit the form without uploading the image.
> 2. Also could you fix the text-overflow, we have lots of free space.
> 
> image

sure, will try to fix this tonight

**Comment by JakubChorzepa** [comment]
> > > Can we fix this problem?
> > > When we add 5 tags. it should show one error message. It keeps adding error messages as we click. I think when we have an error displaced we need to reset the current error toast and show the new one
> > > image
> > 
> > 
> > i think this is the standard way of displaying error toast.
> 
> Is that so 🤔 . What do you say @Yazdun @JakubChorzepa

I think it could be fixed like this proposed idea 
https://forum.obsidian.md/t/debounce-duplicate-notice-toast-popup-notifications/16502

**Comment by rohitdasu** [comment]
@leewan-09 @JakubChorzepa i've updated the PR. please have a look at it.

**Review by leewan-09 — CHANGES_REQUESTED** [review]
1. Add a max limit to the **title**.
2. ON submit "Submit button" should be disabled and show a loader Icon. Current when the user press 2 to 3 time. the function will run 2-3 so it will create duplicated posts. this should be fixed ASAP

**I CLICKED MANY TIMES. LOOK AT THE RESULT**
image

**Review by JakubChorzepa — APPROVED** [review]
LGTM

**Review by leewan-09 — APPROVED** [review]
LGTM 👍🏾, Great PR

**Pull request merged**